src/WindowsUtils/Injection.cpp

```diff
@@ -0,0 +1,274 @@
+  size_t number_of_bytes_read;
+  if (!ReadProcessMemory(process_handle, absl::bit_cast<LPCVOID>(base_address), &result, sizeof(T),
+                         &number_of_bytes_read)) {
+    return ErrorMessage(
+        absl::StrFormat("Error calling ReadProcessMemory: %s", orbit_base::GetLastErrorAsString()));
+  }
+
+  if (number_of_bytes_read != sizeof(T)) {
+    return ErrorMessage("Number of bytes read is not equal to requested size");
+  }
+
+  return outcome::success();
+}
+
+ErrorMessageOr<std::string> RemoteReadString(HANDLE process_handle, uint64_t base_address) {
+  std::string result;
+  char current_char = 'c';
+  size_t number_of_bytes_read = 0;
+  constexpr size_t read_size = sizeof(current_char);
+
+  // We don't know the string length, but it is null terminated. Read one character at a time.
+  for (size_t i = 0;; ++i) {
+    LPCVOID address = absl::bit_cast<LPCVOID>(base_address + i);
+    if (!ReadProcessMemory(process_handle, address, &current_char, read_size,
+                           &number_of_bytes_read)) {
+      return ErrorMessage(absl::StrFormat("Error calling ReadProcessMemory: %s",
+                                          orbit_base::GetLastErrorAsString()));
+    }
+
+    if (number_of_bytes_read != read_size) {
+      return ErrorMessage("Number of bytes read is not equal to requested size");
+    }
+
+    if (current_char == '\0') break;
+    result.push_back(current_char);
+  }
+
+  return result;
+}
+
+ErrorMessageOr<HANDLE> GetHandleFromPid(uint32_t pid) {
+  HANDLE process_handle = OpenProcess(PROCESS_ALL_ACCESS, /*bInheritHandle=*/FALSE, pid);
+  if (!process_handle) {
+    return ErrorMessage(absl::StrFormat("OpenProcess failed for pid %u: %s", pid,
+                                        orbit_base::GetLastErrorAsString()));
+  }
+  return process_handle;
+}
+
+ErrorMessageOr<void> ValidatePath(std::filesystem::path path) {
+  if (!std::filesystem::exists(path))
+    return ErrorMessage(absl::StrFormat("Path does not exist: %s", path.string()));
+  return outcome::success();
+}
+
+[[nodiscard]] std::vector<uint8_t> ToByteBuffer(std::string_view str) {
+  std::vector<uint8_t> result(str.size() + 1, 0);
+  std::memcpy(result.data(), str.data(), result.size());
+  return result;
+}
+
+ErrorMessageOr<Module> FindModule(uint32_t pid, std::string_view module_name) {
+  std::vector<Module> modules = ListModules(pid);
+  std::vector<Module> result;
+  for (const Module& module : modules) {
+    if (absl::EqualsIgnoreCase(module.name, module_name)) {
+      result.push_back(module);
+    }
+  }
+  if (result.size() == 1) return result[0];
+  if (result.size() > 1)
+    return ErrorMessage(
+        absl::StrFormat("Multiple modules with the name \"%s\" found", module_name));
+  return ErrorMessage("Could not find module in target process");
+}
+
+ErrorMessageOr<Module> FindModule(uint32_t pid, std::string_view module_name,
```

Comment:
This method retries with sleeps so the name should be expanded. Something like `FindModuleWithRetries`.